### PR TITLE
Fix logbook showing user names

### LINKED ISSUE #6326: Logbook doesn't load in 0.112.0 for non-admin users
After update to 0.112.0, the logbook no longer load in the app. it works fine from web browser.

Logger: homeassistant.components.websocket_api.http.connection.139821523434704
Source: components/websocket_api/connection.py:97
Integration: Home Assistant WebSocket API (documentation, issues)
First occurred: 6:44:31 AM (1 occurrences)
Last logged: 6:44:31 AM

Error handling message: Unauthorized

### PR DESCRIPTION
**Pull request by balloob**
## Proposed change

Fixes #6326 

We fetched the user list, but that is an admin-only API. Instead we should fetch the persons, which can be accessed by any user.

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by tjhiwa** [comment]
What would happen when the user has no linked person?

**Comment by balloob** [comment]
It won't show any name as who did it.

**Comment by balloob** [comment]
Changed it now to check person first, then fall back to users for admin users only.

File: src/panels/logbook/ha-panel-logbook.ts [hunk 1]
```diff
@@ -197,13 +198,38 @@ export class HaPanelLogbook extends LitElement {
...
+    }
+    for (const person of persons.config) {
+      if (person.user_id) {
+        userIdToName[person.user_id] = person.name;
+      }
+    }
+
+    // Process users
+    if (userProm) {
+      const users = await fetchUsers(this.hass);
```
Comment (bramkragten) [inline src/panels/logbook/ha-panel-logbook.ts:224]:
```suggestion
      const users = await userProm;
```

File: src/panels/logbook/ha-panel-logbook.ts [hunk 2]
```diff
@@ -197,13 +198,38 @@ export class HaPanelLogbook extends LitElement {
...
+    const userIdToName = {};
+
+    // Start loading all the data
+    const personProm = fetchPersons(this.hass);
+    const userProm = this.hass.user!.is_admin && fetchUsers(this.hass);
+
+    // Process persons
+    const persons = await personProm;
+
+    for (const person of persons.storage) {
```
Comment (bramkragten) [inline src/panels/logbook/ha-panel-logbook.ts:211]:
```suggestion
    for (const person of [...persons.storage, ...persons.config]) {
```

File: src/panels/logbook/ha-panel-logbook.ts [hunk 3]
```diff
@@ -197,13 +198,38 @@ export class HaPanelLogbook extends LitElement {
...
+      if (person.user_id) {
+        userIdToName[person.user_id] = person.name;
+      }
+    }
+    for (const person of persons.config) {
+      if (person.user_id) {
+        userIdToName[person.user_id] = person.name;
+      }
+    }
```
Comment (bramkragten) [inline src/panels/logbook/ha-panel-logbook.ts:220]:
```suggestion
```

**Review by bramkragten — APPROVED** [review]

File: src/panels/logbook/ha-panel-logbook.ts (hunk 2 above)
Comment (balloob) [reply src/panels/logbook/ha-panel-logbook.ts:211]: With that we're making slower code (copying things) to make code cleaner

**Pull request merged**